### ISSUE TITLE
Is nodeIntegration required?

### ISSUE THREAD

**Issue by Mintonne** [nklayman/vue-cli-plugin-electron-builder#610]
Hey,

I have been reading up on Electron best practices and the [official docs](https://electronjs.org/docs/tutorial/security#2-do-not-enable-nodejs-integration-for-remote-content) recommend turning off `nodeIntegration` for any renderer that loads remote content.

In my app, I use a preload script to expose any node modules that I need in the renderer process. Everything else is handled via `IPC`.

`nodeIntegration` is not required in my app. Turning it off , however, breaks the app. It just doesn't launch.

### Error message

![image](https://user-images.githubusercontent.com/39894801/71588633-4473fd00-2b33-11ea-8e90-d13961f02bc3.png)

### Root cause

![image](https://user-images.githubusercontent.com/39894801/71588840-04f9e080-2b34-11ea-83d8-c157e5dae131.png)

I tried using a new project, disabled `nodeIntegration` and the error still appeared. There are few issues on Github about this and the solution is always 

> Enable nodeIntegration

Is this a bug or expected behavior?

**Comment by cawa-93** [comment]
Same question

**Comment by sabifa** [comment]
Noticed the same when I tried to turn it off

**Comment by nklayman** [comment]
Didn't realize this many people wanted this, thanks for letting me know. I think the issue is because I set the webpack target to `electron-renderer`. What do you guys think about adding a config option that will leave the target as default, allowing you to disable `nodeIntegration`?

**Comment by nklayman** [comment]
Also, you will have to do [this](https://github.com/electron-userland/spectron#node-integration) for spectron testing.

**Comment by cawa-93** [comment]
@nklayman I don't really understand What is the difference between a `electron-main`, a `electron-renderer` or `electron-preload`? Perhaps it makes sense to set the exact goal for each entry point?

**Comment by Mintonne** [comment]
@cawa-93 https://webpack.js.org/configuration/target/

@nklayman I don't think setting it to default will fix it. The different entry points need different targets.

* main.js (Vue) - web (default)
* backgound.js (Electron) - electron-main

There is also the preload script that can contain `require` statements but it is located in a Vue folder.

**Comment by cawa-93** [comment]
Maybe introduce some configuration like this:
```js
// vue.config.js
module.exports = {
  pluginOptions: {
    electronBuilder: {
      entryPoints: {
        main: {path: ... , target: ...},
        preload: {path: ... , target: ...},
        renderer: {path: ... , target: ...},
      },
      builderOptions: {
        // options placed here will be merged with default configuration and passed to electron-builder
      }
    }
  }
}
```

**Comment by nklayman** [comment]
`nodeIntegration` is now disabled by default on the [v2.0.0 beta](https://github.com/nklayman/vue-cli-plugin-electron-builder/releases/tag/v2.0.0-beta.0), and can be configured.

**Comment by Sparkenstein** [comment]
that sounds great. is preload added by default? 
for the reference, [secure-electron-template](https://github.com/reZach/secure-electron-template) Is one project in react that tries to collect all secure practices for electron in one place. I think having a default example for IPC would help as well.

**Comment by nklayman** [comment]
@Sparkenstein I created #742 to collect some security recommendations that can be implemented in v2.0 and described in a new documentation section. Can you please repost your comment there for better tracking? Thanks!

**Comment by Barrosy** [comment]
> @nklayman I don't really understand What is the difference between a `electron-main`, `electron-renderer` or `electron-preload`? Perhaps it makes sense to set the exact target for each entry point?

I am aware it is quite a while back since you asked this question, however, for any still remaining with questions, more can be found at the documentation page of Electron. [Here](https://www.electronjs.org/docs/latest/tutorial/quick-start#access-nodejs-from-the-renderer-with-a-preload-script) you can find the explanation of what is meant by preloading. [Here](https://www.electronjs.org/docs/latest/tutorial/quick-start#bonus-add-functionality-to-your-web-contents) you can find the explanation of what is meant by renderer. As for main, it would be quite self explanatory as this is where your main Electron related logic should live. This would be your `background.js`. I hope this clears up a bit about your questions.
